watertap/examples/flowsheets/RO_with_energy_recovery/monte_carlo_sampling_RO_ERD.py

```diff
@@ -80,38 +74,79 @@ def run_parameter_sweep(
     set_operating_conditions(m, water_recovery=0.5, over_pressure=0.3, solver=solver)
     initialize_system(m, solver=solver)
 
-    # Simulate once outside the parameter sweep to ensure everything is appropriately initialized
-    solve(m, solver=solver)
-
     # Check if we need to read in the default model values from a file
     if read_model_defauls_from_file:
         set_defaults_from_yaml(m, defaults_fname)
 
-    # Run the parameter sweep study using num_samples randomly drawn from the above range
-    num_samples = 10
+    return m
+
+
+def build_sweep_params(
+    m,
+    use_LHS=False,
+    sweep_params_fname="mc_sweep_params.yaml",
+    read_sweep_params_from_file=False,
+    num_samples=10,
+):
 
     # Define the sampling type and ranges for three different variables
     if read_sweep_params_from_file:
         sweep_params = get_sweep_params_from_yaml(m, sweep_params_fname)
     else:
         sweep_params = get_sweep_params(m, num_samples, use_LHS=use_LHS)
 
-    # Define the outputs to be saved
+    return sweep_params
+
+
+def build_outputs(m, sweep_params):
     outputs = {}
     outputs["EC"] = m.fs.costing.specific_energy_consumption
     outputs["LCOW"] = m.fs.costing.LCOW
+    return outputs
+
+
+def run_parameter_sweep(
+    csv_results_file_name=None,
+    h5_results_file_name=None,
+    seed=None,
+    use_LHS=False,
+    sweep_params_fname="mc_sweep_params.yaml",
+    read_sweep_params_from_file=False,
+    read_model_defauls_from_file=False,
+    defaults_fname="default_configuration.yaml",
+):
+
+    # Set up the solver
+    solver = get_solver()
+
+    # Simulate once outside the parameter sweep to ensure everything is appropriately initialized
+    solve(build_model(), solver=solver)
```

Comment:
This isn't really needed -- and since the model is rebuilt anyways the comment is misleading.

Reply:
Updated the PR to remove this.